manta-crypto/src/arkworks/groth16.rs

```diff
@@ -524,22 +523,21 @@ where
     }
 }
 
-impl<E> Input<Groth16<E>> for bool
-where
-    E: PairingEngine,
-{
-    #[inline]
-    fn extend(&self, input: &mut Vec<E::Fr>) {
-        input.push((*self).into());
-    }
+/// Implements [`Input`] over [`Groth16`] for `$type` that can convert to a field element.
+macro_rules! public_input_impl {
+    ($($type:tt),* $(,)?) => {
+        $(
+            impl<E> Input<Groth16<E>> for $type
+            where
+                E: PairingEngine,
+            {
+                #[inline]
+                fn extend(&self, input: &mut Vec<E::Fr>) {
+                    input.push((*self).into());
+                }
+            }
+        )*
+    };
 }
 
-impl<E> Input<Groth16<E>> for u128
-where
-    E: PairingEngine,
-{
-    #[inline]
-    fn extend(&self, input: &mut Vec<E::Fr>) {
-        input.push((*self).into());
-    }
-}
+public_input_impl!(bool, u8, u16, u32, u64, u128);
```

Comment:
Is there any reason why signed integers can't be made into field elements?

Reply:
We just don't want to think about signed integers right now and figure out how arkworks interprets them, we can add them later but we probably won't be using them.